### PR TITLE
Remove Max value from enums and update some

### PR DESCRIPTION
**Pull request by xPaw**

### CODE REVIEW

**Comment by yaakov-h** [comment]
```
Build FAILED.
Types\SteamID.cs(462,99): error CS0117: 'EAccountType' does not contain a definition for 'Max' [C:\projects\steamkit\SteamKit2\SteamKit2\SteamKit2.csproj]
Types\SteamID.cs(465,101): error CS0117: 'EUniverse' does not contain a definition for 'Max' [C:\projects\steamkit\SteamKit2\SteamKit2\SteamKit2.csproj]
    0 Warning(s)
    2 Error(s)
Time Elapsed 00:00:27.28
```
😢

File: SteamKit2/SteamKit2/Types/SteamID.cs [hunk 1]
```diff
@@ -459,10 +459,10 @@ public bool IsValid
         {
             get
             {
-                if ( this.AccountType <= EAccountType.Invalid || this.AccountType >= EAccountType.Max )
+                if ( this.AccountType <= EAccountType.Invalid || this.AccountType > EAccountType.AnonUser )
```
Comment (JustArchi) [inline SteamKit2/SteamKit2/Types/SteamID.cs:462]: This is prone to errors in the future, especially if the enum gets extended, it's probably better to use a combination of `!Enum.IsDefined() || this.AccountType == EAccountType.Invalid` or if you don't want the performance cost with it, explicitly extract it to some place that tracks valid account types.

Same below.

Comment (xPaw) [reply SteamKit2/SteamKit2/Types/SteamID.cs:462]: If either of these enums change, SteamID code will require changes to handle them regardless.

Comment (yaakov-h) [reply SteamKit2/SteamKit2/Types/SteamID.cs:462]: Can we add a unit test then that these values are the highest values, with a comment to update the SteamID code if it ever changes?

Comment (xPaw) [reply SteamKit2/SteamKit2/Types/SteamID.cs:462]: Done.

**Pull request merged**